pkg/gardener/shoot/extender/extensions/network_filter.go

```diff
@@ -0,0 +1,99 @@
+package extensions
+
+import (
+	"encoding/json"
+	gardener "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryRuntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+const NetworkFilterType = "shoot-networking-filter"
+
+func NewNetworkFilterExtension(filter imv1.Filter) (*gardener.Extension, error) {
+	disabled := isNetworkingFilterDisabled(filter)
+
+	networkingFilterExtension := gardener.Extension{
+		Type:     NetworkFilterType,
+		Disabled: &disabled,
+	}
+
+	if isIngressBlackholingEnabled(filter) {
+		ingressFilterConfig := filter.Ingress
+
+		var filterList []Filter
+		for _, staticIP := range ingressFilterConfig.StaticIPs {
+			var filter = Filter{
+				Network: *staticIP,
+				Policy:  PolicyBlockAccess,
+			}
+			filterList = append(filterList, filter)
+		}
+
+		filterProviderConfig := Configuration{
+			TypeMeta: metav1.TypeMeta{},
+			EgressFilter: &EgressFilter{
+				BlackholingEnabled: ingressFilterConfig.Enabled,
+				StaticFilterList:   filterList,
+			},
+		}
+
+		providerJson, encodingErr := json.Marshal(filterProviderConfig)
+		if encodingErr != nil {
+			return nil, encodingErr
+		}
+
+		networkingFilterExtension.ProviderConfig = &apimachineryRuntime.RawExtension{Raw: providerJson}
+	}
+
+	return &networkingFilterExtension, nil
+}
+
+func isNetworkingFilterDisabled(filter imv1.Filter) bool {
+	return !filter.Egress.Enabled ||
+		(filter.Ingress != nil && !filter.Ingress.Enabled)
```

Comment:
This line will disable extension in the case when `filter.Egress.Enabled == true` AND  `filter.Ingress.Enabled == false`. Is is really intended? As a result we will get 

```
  extensions:
    - type: shoot-networking-filter
      providerConfig:
        egressFilter:
          blackholingEnabled: true
      disabled: true
```

Reply:
I simplified that ( f0034675be925b5cc5dfdf5073f3ebc6660c6b09 ) so it relies only on ` filter.Egress.Enabled`